mypyc/lib-rt/misc_ops.c

```diff
@@ -669,9 +669,71 @@ CPy_Super(PyObject *builtins, PyObject *self) {
     return result;
 }
 
+static bool import_single(PyObject *mod_id,
+                                  PyObject *as_name,
+                                  PyObject **mod_static,
+                                  PyObject *globals_base,
+                                  PyObject *globals) {
+    if (*mod_static == Py_None) {
+        CPyModule *mod = PyImport_Import(mod_id);
+        if (mod == NULL) {
+            return false;
+        }
+        *mod_static = mod;
+    }
+
+    if (as_name == Py_None) {
+        as_name = mod_id;
+    }
```

Comment:
I'm happy to remove this if clause by moving the defaulting to build time. I just implemented it this way first and IMO it makes the generated C a little easier to read (as names stand out more).